examples/graphql-dataloader/graphql_dataloader.md

```diff
@@ -0,0 +1,22 @@
+# GraphQL service - Dataloader
+
+The Ballerina GraphQL module provides the capability to batch and cache data fetching from data sources using the `graphql.dataloader` submodule. To leverage this functionality in a GraphQL service, users must register data loaders through the `graphql:Context` object and implement the corresponding prefetch method logic and resolver method logic. A prefetch method, in the context of Ballerina GraphQL, is a method that is invoked before the actual resolver method. By default, the prefetch method name follows the convention of the prefix `pre`, followed by the resolver method name. The use of `graphql.dataloader` helps users avoid excessive data fetching, effectively addressing the GraphQL N+1 problem.
```

Comment:
```suggestion
The Ballerina GraphQL module provides the capability to batch and cache data fetching from data sources using the `graphql.dataloader` submodule. To leverage this functionality in a GraphQL service, you must register data loaders through the `graphql:Context` object and implement the corresponding pre-fetch method logic and resolver method logic. A pre-fetch method in the context of Ballerina GraphQL, is a method that is invoked before the actual resolver method. By default, the pre-fetch method name follows the convention of having the prefix `pre` followed by the resolver method name. The use of the `graphql.dataloader` avoids excessive data fetching, effectively addressing the GraphQL N+1 problem.
```
- Check if it is correct to use `batch` as a verb in the first sentence.
- This paragraph is also too long. Consider breaking it into multiple sentences.